src/NServiceBus.RabbitMQ/Administration/QueueCreator.cs

```diff
@@ -50,5 +53,46 @@ void CreateQueueIfNecessary(IModel channel, string receivingAddress)
 
             routingTopology.Initialize(channel, receivingAddress);
         }
+
+        void BuildDeferralInfrastructure(IModel channel)
+        {
+            const int maxLevel = 32 - 1;
+
+            var bindingKey = "1.#";
+
+            for (int i = maxLevel; i >= 0; i--)
+            {
+                var level = i.ToString("D2");
+                var nextLevel = (i - 1).ToString("D2");
+
+                channel.ExchangeDeclare($"nsb.delay-level-{level}", ExchangeType.Topic, durableMessagesEnabled);
+
+                var arguments = new Dictionary<string, object>();
+                arguments.Add("x-queue-mode", "lazy");
+                arguments.Add("x-message-ttl", ((long)1 << i) * 1000);
```

Comment:
I know that IT people should understand it straightaway, but it took me 15 seconds to recall that `<<` is a shift operator and later on to understand that basically what we are doing is calculating 2^i. However replacing it with `Math.Pow(2, i)` would require casting to long. All in all I wonder if we should not write a little inline comment to express what is it. 

This is my braindump on this matter.